### PR TITLE
Improvements to `autogenerate-bloqs-notebooks-v2.py`

### PR DESCRIPTION
**Pull request by tanujkhattar**
`autogenerate-bloqs-notebooks-v2.py` got stale after the recent reorganization of Bloqs. This PR fixes bugs and adds a notebook for `Hadamard` since it was raising a warning.  

Now, running `autogenerate-bloqs-notebooks-v2.py` actually generates all the correct notebooks and does not raise an error.

### CODE REVIEW

File: qualtran/bloqs/swap_network/swap_network.ipynb
Comment (mpharrigan) [inline qualtran/bloqs/swap_network/swap_network.ipynb:1]: can you revert these changes? The id should generally be preserved

File: dev_tools/qualtran_dev_tools/jupyter_autogen_v2.py [hunk 1]
```diff
@@ -52,17 +52,17 @@ class NotebookSpecV2:
     module: ModuleType
     bloq_specs: List[BloqDocSpec]
     directory: str = field()
-    _path_stem: Optional[str] = None
+    path_stem_name: Optional[str] = None
```
Comment (mpharrigan) [inline dev_tools/qualtran_dev_tools/jupyter_autogen_v2.py:55]: why this change? attrs will strip the leading underscore to accept this as a init arg

**Comment by mpharrigan** [comment]
can you also update the sphinx toc #794

**Comment by tanujkhattar** [comment]
@mpharrigan I've reverted the changes `swap_network.ipynb`. Can I update the sphinx toc in a follow-up PR?

File: dev_tools/qualtran_dev_tools/jupyter_autogen_v2.py (hunk 1 above)
Comment (tanujkhattar) [reply dev_tools/qualtran_dev_tools/jupyter_autogen_v2.py]: That's correct, this is not needed. I've reverted the change. Thanks for flagging.

File: qualtran/bloqs/swap_network/swap_network.ipynb
Comment (tanujkhattar) [reply qualtran/bloqs/swap_network/swap_network.ipynb:1]: done

**Review by mpharrigan — APPROVED** [review]

**Comment by mpharrigan** [comment]
> Can I update the sphinx toc in a follow-up PR?
yep!
